warehouse/accounts/services.py

```diff
@@ -300,13 +300,14 @@ def update_user(self, user_id, **changes):
 
         return user
 
-    def disable_password(self, user_id, reason=None, ip_address="127.0.0.1"):
+    def disable_password(self, user_id, reason=None, request=None):
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏the calls to `user.record_event` ought to fail  with an AttributeError of `request` is None here when calling for `request.remote_addr`

I _think_ that can only happen today where an Admin view doesn't pass `request` (or really `ip_address`) when calling this function. It appears that you've updated the caller, so maybe it's time to make `request` non-optional and remove the default?

Reply:
6801f5541